core/src/HIP/Kokkos_HIP_ParallelReduce_Range.hpp

```diff
@@ -240,14 +242,14 @@ class ParallelReduce<CombinedFunctorReducerType, Kokkos::RangePolicy<Traits...>,
             m_policy.space()
                 .impl_internal_space_instance()
                 ->m_deviceProp.maxSharedMemoryPerMultiProcessor;
-        Kokkos::Impl::throw_runtime_exception(
-            std::string("Kokkos::Impl::ParallelReduce< HIP > could not find a "
-                        "valid execution configuration: your kernel requires " +
-                        std::to_string(shared_memory_required) +
-                        " bytes of shared memory per block but only " +
-                        std::to_string(shared_memory_available) +
-                        " bytes per block are available."));
+        printf(
+            "Kokkos::Impl::ParallelReduce< HIP > could not find a valid "
+            "execution configuration: your kernel requires %u bytes of shared "
+            "memory per block but only %u bytes per block are available.",
+            shared_memory_required, shared_memory_available);
+        KOKKOS_ASSERT(false);
```

Comment:
If `KOKKOS_ENFORCE_CONTRACTS` is true, does this assertion trigger? Besides the fact that I didn't remember that we had `KOKKOS_ENFORCE_CONTRACTS` I feel like it should actually abort or whatever here, and not just print a message if something is wrong.

Reply:
> If KOKKOS_ENFORCE_CONTRACTS is true, does this assertion trigger?

Yes, it does. I copied the code from `KOKKOS_ASSERT`

> I feel like it should actually abort or whatever here, and not just print a message if something is wrong.

What do you mean? We print the message and then we abort. I kept the message because one user asked to have a better error message for `parallel_reduce`.

Reply:
Use `Kokkos::abort()` instead.